test/ir_Corona_test.cpp

```diff
@@ -0,0 +1,1550 @@
+#include "IRsend.h"
+#include "IRsend_test.h"
+#include "gtest/gtest.h"
+
+// Tests for decodeCoronaAc().
+TEST(TestDecodeCoronaAc, RealExample) {
+  IRsendTest irsend(kGpioUnused);
+  IRrecv irrecv(kGpioUnused);
+
+  const uint16_t rawData_On[347] = {
+      3520, 1686,
+      460, 386,  452, 454,  408, 452,  410, 1320, 434, 428,  434, 1296,
+      410, 452,  490, 374,  438, 1292, 434, 430,  434, 428,  438, 424,
+      438, 426,  438, 1266, 464, 1266, 462, 402,  462, 1266, 464, 400,
+      462, 1268, 464, 1266, 462, 1266, 464, 1268, 486, 376,  462, 402,
+      472, 1256, 488, 378,  514, 346,  464, 1266, 488, 1240, 462, 404,
+      462, 426,  460, 376,  462, 402,  488, 1240, 464, 1266, 462, 402,
+      462, 400,  490, 1240, 462, 1266, 486, 1244, 460, 1268, 462, 1266,
+      462, 1266, 462, 404,  458, 1270, 458, 1268, 462, 400,  464, 428,
+      434, 402,  460, 402,  462, 398,  462, 1268, 462, 404,  460, 402,
+      460, 1266, 466, 1264,
+      462, 10808,
+      3502, 1686,
+      462, 400,  462, 400,  462, 400,  464, 1266, 462, 404,  458, 1266,
+      464, 402,  460, 402,  462, 1266, 464, 404,  458, 402,  460, 400,
+      488, 374,  464, 1266, 462, 1266, 466, 398,  462, 1268, 462, 400,
+      518, 1208, 438, 1292, 440, 426,  488, 1242, 466, 1264, 460, 402,
+      490, 1238, 490, 1238, 490, 1242, 514, 1212, 464, 1268, 490, 1238,
+      464, 1266, 490, 1238, 462, 406,  458, 400,  464, 398,  464, 402,
+      488, 374,  486, 376,  460, 402,  488, 372,  464, 1268, 462, 1264,
+      464, 1266, 462, 1268, 462, 1266, 466, 1266, 490, 1242, 458, 1264,
+      516, 348,  462, 404,  460, 404,  462, 398,  516, 344,  462, 400,
+      464, 402,  488, 376,
+      460, 10810,
+      3526, 1658,
+      462, 404,  460, 400,  488, 374,  464, 1264, 462, 402,  464, 1266,
+      462, 402,  486, 374,  462, 1266, 546, 318,  462, 400,  464, 400,
+      462, 402,  460, 1268, 466, 1262, 492, 374,  516, 1212, 464, 400,
+      462, 1266, 516, 1212, 464, 402,  462, 424,  462, 1242, 460, 1268,
+      488, 1242, 462, 1268, 490, 1236, 466, 1264, 462, 1268, 464, 1266,
+      460, 1268, 464, 1266, 462, 338,  528, 402,  486, 374,  462, 402,
+      460, 402,  490, 374,  458, 406,  458, 402,  488, 1240, 490, 1238,
+      462, 1268, 464, 1268, 490, 1238, 488, 1242, 462, 1264, 464, 1270,
+      462, 398,  436, 426,  460, 406,  458, 404,  434, 424,  464, 400,
+      462, 400,  460, 402,
+      462};  // UNKNOWN 94D81276 * ON
+  const uint8_t expectedState_On[kCoronaAcStateLength] = {
+      0x28, 0x61, 0x3D, 0x19, 0xE6, 0x37, 0xC8,
+      0x28, 0x61, 0x6D, 0xFF, 0x00, 0xFF, 0x00,
+      0x28, 0x61, 0xCD, 0xFF, 0x00, 0xFF, 0x00};
+  irsend.begin();
+  irsend.reset();
+  irsend.sendRaw(rawData_On, 347, 38);
+  irsend.makeDecodeResult();
+
+  ASSERT_TRUE(irrecv.decode(&irsend.capture));
+  ASSERT_EQ(decode_type_t::CORONA_AC, irsend.capture.decode_type);
+  ASSERT_EQ(kCoronaAcBits, irsend.capture.bits);
+  EXPECT_STATE_EQ(expectedState_On, irsend.capture.state, irsend.capture.bits);
+
```

Comment:
I recommend adding a test of `resultAcToString()` here.
e.g.
```
  EXPECT_EQ("Power: On, Mode: ...", IRAcUtils::resultAcToString(&irsend.capture));
```

Reply:
I was thinking about that as well, but that requires creating an ac instance(?), will do.

Reply:
No, it doesn't require it. It's a `static` method of the class, meaning it can be called with out an instance of the class object. It does all the processing for you internal to itself.